pkg/ebpf/c/common/context.h

```diff
@@ -208,11 +231,30 @@ statfunc int init_tailcall_program_data(program_data_t *p, void *ctx)
     return 1;
 }
 
-// use this function for programs that send more than one event
-statfunc void reset_event_args(program_data_t *p)
+// use this function in programs that send more than one event
+statfunc bool reset_event(event_data_t *event, u32 event_id)
 {
-    p->event->args_buf.offset = 0;
-    p->event->args_buf.argnum = 0;
+    event->context.eventid = event_id;
+    event->args_buf.offset = 0;
+    event->args_buf.argnum = 0;
+    event->config.submit_for_policies = ~0ULL;
+
+    event_config_t *event_config = get_event_config(event_id, event->context.policies_version);
+    if (event_config == NULL)
+        return false;
```

Comment:
I meant: only touch event fields if there actually is a config for that event, like:

```c
statfunc bool reset_event(event_data_t *event, u32 event_id)
{
    // check first
    event_config_t *event_config = get_event_config(event_id, event->context.policies_version);
    if (event_config == NULL)
        return false;

    // reset if there's a config for it.
    event->context.eventid = event_id;
    reset_event_args_buf(event);
    event->config.submit_for_policies = ~0ULL;
    ...
```

Reply:
I don't know about this change.
These fields not depend on the event config, and cost almost nothing (just simple assignments).
In case the caller don't check the return value, returning from reset_event after doing these 4 assignments makes the reset valid, and matched_policies is set to match all policies.